### PR TITLE
Adds run command -> install and launch extension

### PR DESCRIPTION
**Pull request by kumar303**

### CODE REVIEW

**Comment by kumar303** [comment]
This pattern of adding `adapter.js` to the tests is something that will go away in https://github.com/mozilla/web-ext/issues/87 . It was just a hack.

**Comment by rpl** [comment]
@kumar303 I think that we need to add to the cli a new "--firefox-bin" option (as a non mandatory option), because the firefox binary auto-discovery sometimes fails (e.g. on my ubuntu it decides that the binary should be /usr/lib64/firefox doesn't even exist on my system)

File: src/firefox/index.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    })
+    .catch(onlyErrorsWithCode('ENOENT', () => {
+      console.log(`Creating extensions directory: ${profile.extensionsDir}`);
+      return fs.mkdir(profile.extensionsDir);
+    }))
+    .then(() => {
+      let readStream = nodeFs.createReadStream(extensionPath);
+      let id = manifestData.applications.gecko.id;
+
+      // TODO: also support copying a direcotry of code to this
```
Comment (rpl) [inline src/firefox/index.js]: nit, typo: direcotry -> directory

**Comment by kumar303** [comment]
> I think that we need to add to the cli a new "--firefox-bin" option

Yep, adding that in https://github.com/mozilla/web-ext/issues/71

File: package.json [hunk 2]
```diff
@@ -37,6 +37,8 @@
     "babel-polyfill": "6.6.1",
     "es6-error": "2.0.2",
     "es6-promisify": "3.0.0",
+    "firefox-profile": "^0.3.11",
```
Comment (rpl) [inline package.json]: @kumar303 This should be turned into an exact version match

Comment (kumar303) [inline package.json]: d'oh! Good catch.

File: src/firefox/index.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+        firefox.on('error', (error) => {
+          // TODO: show a nice error when it can't find Firefox.
+          // if (/No such file/.test(err) || err.code === 'ENOENT') {
+          console.log(`Firefox error: ${error}`);
+          throw error;
+        });
+
+        firefox.stderr.on('data', (data) => {
+          console.log(`stdout: ${data.toString().trim()}`);
```
Comment (rpl) [inline src/firefox/index.js]: @kumar303 shouldn't this be stderr? (e.g. ```console.log(`stderr: ...`);```)

why not using console.error here to route these messages to the stderr?

File: src/firefox/index.js [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        });
+
+        firefox.stdout.on('data', function(data) {
+          console.log(`stderr: ${data.toString().trim()}`);
```
Comment (rpl) [inline src/firefox/index.js]: @kumar303 if I wasn't wrong in the above comment, then this should be console.log(`stdout: ...`);

File: src/firefox/index.js (hunk 3 above)
Comment (kumar303) [inline src/firefox/index.js]: oh yeah, heh, good catch. I also plan to fix up all the logging in https://github.com/mozilla/web-ext/issues/33

File: src/firefox/index.js (hunk 4 above)
Comment (kumar303) [inline src/firefox/index.js]: ha! I don't eve know how that happened.

**Comment by rpl** [comment]
@kumar303 great! this PR looks ready.

I'm so happy that I can now run a webextension addon without leaving the command line :-)

**Comment by kumar303** [comment]
Thanks for the review! Yes, definitely let me know how it works for you. There are, of course, a lot of missing features (a lot of issues filed).

**Pull request merged**